topo/node/juniper/juniper.go

```diff
@@ -427,10 +493,42 @@ func defaults(pb *tpb.Node) *tpb.Node {
 			Name: "default_cptx_node",
 		}
 	}
-	if pb.Constraints == nil {
-		pb.Constraints = map[string]string{
-			"cpu":    "8",
-			"memory": "8Gi",
+	if pb.Model == "" {
+		pb.Model = ModelCPTX
+	}
+	if pb.Config == nil {
+		pb.Config = &tpb.Config{}
+	}
+	switch pb.Model {
+	case ModelNCPTX:
+		if pb.Constraints == nil {
+			pb.Constraints = map[string]string{
+				"cpu":    "4",
+				"memory": "4Gi",
+			}
+		}
+		if len(pb.Config.GetCommand()) == 0 {
+			pb.Config.Command = []string{
+				"/sbin/cevoCntrEntryPoint",
+			}
+		}
+		if pb.Config.Image == "" {
+			pb.Config.Image = "ncptx:latest"
+		}
+	default:
+		if pb.Constraints == nil {
+			pb.Constraints = map[string]string{
+				"cpu":    "8",
+				"memory": "8Gi",
+			}
+		}
+		if len(pb.Config.GetCommand()) == 0 {
+			pb.Config.Command = []string{
+				"/entrypoint.sh",
+			}
+		}
+		if pb.Config.Image == "" {
+			pb.Config.Image = "cptx:latest"
 		}
 	}
 	if pb.Services == nil {
```

Comment:
please add p4rt to this list if it works

Reply:
p4rt was already present but does not work yet - removed it instead.

Reply:
does it only not work for ncptx or also cptx?

if its only ncptx then change the defaults per model.

The presubmit failed because p4rt is no longer found on cptx due to the removal from defaults

Reply:
> does it only not work for ncptx or also cptx?
> 
> if its only ncptx then change the defaults per model.
> 
> The presubmit failed because p4rt is no longer found on cptx due to the removal from defaults

Apparently, p4rt does not work for both cptx and ncptx. We may need to remove it from vendors_test.go.

Reply:
Do let me know if you want to me push another commit commenting out p4rt for cptx: https://github.com/openconfig/kne/blob/c4e15946368bea4d681ce7f0dc88c421df36856a/cloudbuild/vendors/vendors_test.go#L81

Reply:
Per my discussion with the folks, the p4rt is not yet qualified/supported on cptx and hence ncptx. There are some bits that may need enhancement. You are probably able to dial because basics are in place but there might be bugs. I will comment it out and re-enable it later when things change.